x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/oidc/OpenIdConnectRealm.java

```diff
@@ -414,17 +410,21 @@ static ClaimParser forSetting(Logger logger, OpenIdConnectRealmSettings.ClaimSet
                             Object claimValueObject = claims.getClaim(claimName);
                             List<String> values;
                             if (claimValueObject == null) {
-                                values = Collections.emptyList();
+                                values = List.of();
                             } else if (claimValueObject instanceof String) {
-                                values = Collections.singletonList((String) claimValueObject);
+                                values = List.of((String) claimValueObject);
                             } else if (claimValueObject instanceof List) {
                                 values = (List<String>) claimValueObject;
                             } else {
                                 throw new SettingsException("Setting [" + RealmSettings.getFullSettingKey(realmConfig, setting.getClaim())
-                                    + " expects a claim with String or a String Array value but found a "
+                                    + " expects a claim with String or a String List value but found a "
```

Comment:
I'll defer to @jkakavas on this, but I think the original message was correct. This is refering to claims in the OIDC JSON, and the JSON term for a multi-valued element is an Array, not a List.

Reply:
Okay, I'll revert this.